builtin/providers/aws/resource_aws_autoscaling_metrics_collection.go

```diff
@@ -0,0 +1,140 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsAutoscalingMetric() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAutoscalingMetricCreate,
+		Read:   resourceAwsAutoscalingMetricRead,
+		Update: resourceAwsAutoscalingMetricUpdate,
+		Delete: resourceAwsAutoscalingMetricDelete,
+
+		Schema: map[string]*schema.Schema{
+			"autoscaling_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"metrics": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"granularity": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAutoscalingMetricCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).autoscalingconn
+
+	props := &autoscaling.EnableMetricsCollectionInput{
+		AutoScalingGroupName: aws.String(d.Get("autoscaling_group_name").(string)),
+		Granularity:          aws.String(d.Get("granularity").(string)),
+	}
+
+	if v, ok := d.GetOk("metrics"); ok {
+		props.Metrics = expandStringList(v.(*schema.Set).List())
+	}
+
+	log.Printf("[INFO] Enabling metrics collection for the ASG: %s", d.Id())
+	_, err := conn.EnableMetricsCollection(props)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("autoscaling_group_name").(string))
```

Comment:
Let's get/generate a unique name here for the ID here. I envision someone(s) pulling an ASG name out of a module and adding metrics collections in other spots and I worry about a name conflict. Sound reasonable? Specifically because the `name` is coming from an external resource